### PR TITLE
Update User profile interfaces to async

### PR DESCRIPTION
**Pull request by ts-leojoseph-r**
# Description
Update the following interfaces into async

- getUserName()
- GetProfilePhoto()

## Links
MINI-2121

# Checklist
- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `fastlane ci` without errors

### CODE REVIEW

**Review by Climbatize — APPROVED** [review]
Looks good to me. I just have one suggestion. You could keep the old methods without deprecating them. Here is a suggestion you could apply to all your methods to keep retro-compatibility:

keep the delegate as is and implement it in host app (you can even simplify the one that was in place):
```swift
func getUserName() -> String? {
        return getProfileSettings()?.displayName
}
```
and in the public extension of MiniAppUserInfoDelegate as a default implementation for the new method make a call to the old one:

```swift
func getUserName(completionHandler: @escaping (Result<String?, MASDKError>) -> Void?) {
        guard let userName = getUserName() else {
            return completionHandler?(.failure(.unknownError(domain: "Unknown Error", code: 1, description: "Failed to retrieve User name")))
        }
        completionHandler(.success(userName)) ?? username
    }
```

**Comment by ts-leojoseph-r** [comment]
> Looks good to me. I just have one suggestion. You could keep the old methods without deprecating them. Here is a suggestion you could apply to all your methods to keep retro-compatibility:
> 
> keep the delegate as is and implement it in host app (you can even simplify the one that was in place):

Thanks for the suggestion, but it is the Host app that requested to make this Async, keeping the interface the same and just changing the default extension wouldn't be a solution that they would need.

File: MiniApp/Classes/JavascriptBridge/MiniAppScriptMessageHandler.swift [hunk 1]
```diff
@@ -167,26 +167,48 @@ internal class MiniAppScriptMessageHandler: NSObject, WKScriptMessageHandler {
...
+            hostAppMessageDelegate?.getProfilePhoto { (result) in
+                switch result {
+                case .success(let response):
+                    guard let profilePhoto = response else {
+                        self.executeJavaScriptCallback(responseStatus: .onError, messageId: callbackId, response: getMiniAppErrorMessage(MiniAppErrorType.hostAppError))
+                        return
+                    }
+                    self.executeJavaScriptCallback(responseStatus: .onSuccess, messageId: callbackId, response: profilePhoto)
+                case .failure(let error):
+                    if !error.localizedDescription.isEmpty {
```
Comment (corycaywood) [inline MiniApp/Classes/JavascriptBridge/MiniAppScriptMessageHandler.swift:202]: It seems like there's some code duplication with the `.failure` case. Most of the functions in this class do the same thing on failure. Is there a way to extract this into another function?

**Review by corycaywood — APPROVED** [review]
Looks good 👍  Just left a suggestion.

File: MiniApp/Classes/JavascriptBridge/MiniAppScriptMessageHandler.swift (hunk 1 above)
Comment (ts-leojoseph-r) [reply MiniApp/Classes/JavascriptBridge/MiniAppScriptMessageHandler.swift]: Done

**Pull request merged**